src/components/module-card/index.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+
+import { WaveclassTheme } from '../../layout';
+import { ModuleCardProps } from './types';
+
+import * as s from './styles';
+import { Typography } from '../typography';
+
+import {
+	TfiAngleDown,
+	TfiAngleUp,
+	TfiLock,
+	TfiTime,
+} from 'react-icons/tfi';
+import { ProgressBar } from '../progress-bar';
+import { GridItem } from '../grid-item';
+
+export const ModuleCard = ({
+	name,
+	pos,
+	totaltime,
+	percentProgress,
+	available,
+	availableMessage,
+	expanded,
+}: ModuleCardProps): JSX.Element => {
+	return (
+		<>
+			<WaveclassTheme>
+				<s.ModuleCardContainer
+					isComplete={percentProgress >= 100}
+				>
+					<s.ModuleMainDetails>
+						<s.ModulePositionContainer>
+							<Typography
+								variant="h4-bold"
+								text={`${pos}`}
+							/>
+						</s.ModulePositionContainer>
+
+						<s.ModuleDetailsContainer>
+							<Typography
+								variant="h4-bold"
+								text={name}
+							/>
+
+							<s.ModuleDetailsFooter>
+								<s.ModuleDurationDisplay>
+									<TfiTime size={12} />
+									<Typography
+										variant="small-regular"
+										text={totaltime}
+									/>
+								</s.ModuleDurationDisplay>
+
+								<GridItem variant="col-8">
+									<ProgressBar
+										percentProgress={
+											percentProgress
+										}
+									/>
+								</GridItem>
+							</s.ModuleDetailsFooter>
+						</s.ModuleDetailsContainer>
+					</s.ModuleMainDetails>
+
+					<s.ExpanderContainer>
```

Comment:
O uso do operador `||` está causando um bug grave aqui, pois mesmo se `available` for falso, vai exibir a flecha, pois "se estiver disponível OU não estiver expandido". Assim, você libera a um aluno um módulo mesmo sem ele ter permissão de acesso.

Minha sugestão:

```
<s.ExpanderContainer>
  {available ? (
	  <s.ExpanderIconContainer>
	  {!expanded ? (
		  
			  <TfiAngleDown
				  size={16}
			  />
	  ) : (
			  <TfiAngleUp size={16} />
	  )}
	  </s.ExpanderIconContainer>
  ) : (
	  <s.ExpanderIconContainer>
		  <TfiLock size={16} />
		  <Typography
			  variant="small-regular"
			  text={availableMessage}
		  />
	  </s.ExpanderIconContainer>
  )}
  </s.ExpanderContainer>
```

Isolando a lógica do `available` e do `expanded`, você garante que eles não se misturem.